environment.yml

```diff
@@ -1,13 +1,13 @@
-name: GeoNet
+name: geonet-tutorial
 channels:
   - conda-forge
 dependencies:
-  - python=3.10
+  - python[version='>=3.12']
   - cartopy
   - numpy
```

Comment:
is there any recommended version of cartopy and numpy we shall add?

Reply:
I couldn't find any recommendations on the degree to which you should specify module versions. In many cases I've seen people don't specify any versions at all. I know its intended to ensure you have versions that aren't too old and offer the functionality the code writer intends, but maybe also stops someone using newer versions?
If I make the environment now I may have a newer version of `cartopy`, say, than GeoNet when they wrote the code. To avoid that GeoNet would have to specify every module to be exactly what we used. But as I said, there weren't any recommendations I could find about that.

What is there is a bit of a compromise. The key thing is that it should work. If a user has problems they can always ask. I don't see how we can plan for every contingency.

For `cartopy`, I'm pretty sure its not updated that often. I think it is used only in a couple of tutorials, where we draw a map. So might be considered of lesser importance. My build of `geonet-tutorial` gives me 0.24.0 and the latest is 0.24.1.

I was a bit scared of specifying `numpy` as it is very important. I know pandas is closely built on it, so I thought specifying `pandas` would constrain `numpy` to be compatible. I don't think `numpy` is explicitly used that much in the tutorials. My build of `geonet-tutorial` gives me 1.26.4 and the latest that supports python 3.12 is 1.26.0 (I assume they aren't showing the sub-release). It looks as if once you up the python version numpy will go to a newer version.

The biggest change is specifying python 3.12. Ver 3.10 is Oct 2021, ver 3.12 is Oct 2023. Ver 3.13 is latest, release 2 months ago. It just seemed logical to use a newer version - I'm assuming there must be benefits in doing that. My build of `geonet-tutorial` gives me python 3.12.7.

Finally, if someone more knowledgeable can suggest how we should specifiy the environment then we can do it that way.

Reply:
thanks a lot @rumachan it all makes sense, so thanks heaps for taking the time to reply!